### DISCUSSION TITLE
[iOS] Unable to run crossplatform example on real iOS Device

### DISCUSSION THREAD

**Discussion by maxhoerstr** [AvaloniaUI/Avalonia#16006] [Q&A] [answered]
Hello everyone,

I´m trying to run the [crossplatform example](https://docs.avaloniaui.net/docs/get-started/install#install-avalonia-ui-templates) on an iPad that is connected to my Macbook via USB. i did all the provisioning, run the sample app with xCode and added all the information according to the tutorial.  Im using VSCode so i run `dotnet build` and `dotnet run`.  The building is working, however when i try to run the app the following error occours:

`EXEC : error MT1044: A device must be specified using --devname.`

I´m also getting many of these errors:
`2024-06-12 09:21:08.899 mlaunch[14928:125681] Requested but did not find extension point with identifier Xcode.IDEDebugger.VariablesViewQuickLookProvider for extension Xcode.IDEDebugger.SpriteKitQuickLookProvider of plug-in com.apple.IDESpriteKitParticleEditor`

When im changing the runtime identifier and use dotnet run, the emulator starts and i can run the app. Is this problem maybe related to the way i start the application? Do i need Visual Studio for Mac?

Thanks in advance!

**Comment by maxkatz6** [comment, marked as answer]
Apparently you can't simply use "dotnet run" on iOS/Android.
See what options you have here: https://github.com/dotnet/xamarin/issues/26

Also, while Microsoft didn't release iOS/Android compatible .NET extension for VS Code, Avalonia also works with combination of https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit and https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-maui

**Reply by maxhoerstr** [reply]
Thank you very much! The following option started my App on the iPad: `dotnet build -t:Run /p:_DeviceName=<UDID>`
